ui/src/image_annotator_polygon.ts

```diff
@@ -1,41 +1,113 @@
 import { F, S, U } from "h2o-wave"
-import { DrawnPoint, DrawnShape, ImageAnnotatorPoint, Position } from "./image_annotator"
+import { DrawnPoint, DrawnShape, ImageAnnotatorPoint, ImageAnnotatorRect, Position } from "./image_annotator"
 import { ARC_RADIUS } from "./image_annotator_rect"
 
 export class PolygonAnnotator {
   private currPolygonPoints: ImageAnnotatorPoint[] = []
+  private boundaryRect: ImageAnnotatorRect | undefined
   private draggedPoint: ImageAnnotatorPoint | null = null
   private draggedShape: DrawnShape | null = null
+  private ctx: CanvasRenderingContext2D | null
 
-  constructor(private ctx: CanvasRenderingContext2D) {
-    this.ctx.lineWidth = 2
+  constructor(private canvas: HTMLCanvasElement) {
+    this.ctx = canvas.getContext('2d')
+    if (this.ctx) this.ctx.lineWidth = 2
   }
 
   resetDragging() {
+    // Update the boundaries of the polygon when point dragging ends.
+    const draggedPolygon = this.draggedShape?.shape.polygon
+    if (this.draggedPoint && draggedPolygon) {
+      this.boundaryRect = draggedPolygon.boundaryRect
+      this.updateBoundaryRect(this.draggedPoint)
+      draggedPolygon.boundaryRect = this.boundaryRect
+      this.boundaryRect = undefined
```

Comment:
Make pass the `boundaryRect` you want to mutate as a param to `updateBoundary` to avoid these gymnastics.

Side question: Why do we need to update the boundaries when resetting the dragging? Isn't the latest boundary value from onMove enough?

Reply:
>  Isn't the latest boundary value from onMove enough?

We update latest boundaries only when you are moving the whole shape. This is for the case when you are dragging a single point (to re-shape the existing polygon), so you don't have the latest boundary values up to date and we are updating it here.

Reply:
I see, makes sense. Thanks!